src/main/java/com/stuypulse/robot/util/MotorSim.java

```diff
@@ -0,0 +1,125 @@
+package com.stuypulse.robot.util;
+
+import java.util.function.Function;
+
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.math.system.plant.LinearSystemId;
+import edu.wpi.first.util.sendable.Sendable;
+import edu.wpi.first.util.sendable.SendableBuilder;
+import edu.wpi.first.wpilibj.motorcontrol.MotorController;
+import edu.wpi.first.wpilibj.simulation.DCMotorSim;
+
+/**
+ * A simulated motor intended to be a simple replacement for
+ * real motors. Internally uses a LinearSystemSim constructed
+ * based on the given motor type. Holds a reference to an EncoderSim
+ * which can be accessed through .getEncoder().
+ * 
+ * @author Ben Goldfisher
+ */
+public class MotorSim implements MotorController, Sendable {
```

Comment:
It may be easier to let the type of linear system sim depends on the subsystem (position control, velocity control) even though everything has motors. 

Also DCMotorSim is [angular position, angular velocity] when it may make more sens to use a generic system that has output of linear units of height and velocity. 

A DCMotorSim may be preferable for the climber if we can figure out all the parameters because we know how to convert from output rotations to a change in height (it may be a more accurate simulation that doesn't require SysId'ing)